### PR TITLE
Use EFCore.NamingConventions to rewrite the name

### PR DESCRIPTION
**Pull request by ghost**
I'm trying to fix the issues #2131 & #2351 , use **[EFCore.NamingConventions](https://github.com/efcore/EFCore.NamingConventions)** to rewrite the name. Supported naming conventions: Default, SnakeCase, LowerCase, UpperCase, UpperSnakeCase.

Changed:
Add **Volo.Abp.EntityFrameworkCore** reference to **EFCore.NamingConventions**.  
In the **xxxDbContext.cs** , enable rewrite name by the extension methods **NamingConventionsRewriteName** of class **Volo.Abp.EntityFrameworkCore.DbNamingConventionRewriterExtensions.cs**

```
protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)  
        {  
            base.OnConfiguring(optionsBuilder);
     --->       optionsBuilder.NamingConventionsRewriteName(BackgroundJobsDbProperties.DbNamingConvention);
        }
  
        protected override void OnModelCreating(ModelBuilder builder)
        {
            base.OnModelCreating(builder);
            builder.ConfigureBackgroundJobs();
      --->   builder.NamingConventionsRewriteName(BackgroundJobsDbProperties.DbNamingConvention);
        }
```

Use outside the abp framework, set the name convensions at the beginning.

```
AbpCommonDbProperties.DbNamingConvention = "SnakeCase";  
AbpIdentityServerDbProperties.DbNamingConvention = "SnakeCase";
```

### CODE REVIEW

**Comment by hikalkan** [comment]
Thanks @scymen. We will check the PR.

File: framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/EntityFrameworkCore/DbNamingConventionRewriterExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                case nameof(DbNamingConvention.UpperCase):
+                    optionsBuilder.UseUpperCaseNamingConvention(culture);
+                    break;
+                case nameof(DbNamingConvention.UpperSnakeCase):
+                    optionsBuilder.UseUpperSnakeCaseNamingConvention(culture);
+                    break;
+                default:
+                    break;
+            }
+        }
```
Comment (maliming) [inline framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/EntityFrameworkCore/DbNamingConventionRewriterExtensions.cs:61]: You can try to convert `(namingConvention )string`to `DbNamingConvention`, and then reuse `NamingConventionsRewriteName `method.

File: framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/EntityFrameworkCore/DbNamingConventionRewriterExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                    key.SetConstraintName(nameRewriter.RewriteName(key.GetConstraintName()));
+                }
+
+                foreach (var index in entity.GetIndexes())
+                {
+                    index.SetName(nameRewriter.RewriteName(index.GetName()));
+                }
+            }
+
+        }
```
Comment (maliming) [inline framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/EntityFrameworkCore/DbNamingConventionRewriterExtensions.cs:134]: Is it duplicate?
https://github.com/efcore/EFCore.NamingConventions/blob/36a41daa87/EFCore.NamingConventions/Internal/NamingConventionSetPlugin.cs#L15-l40

File: framework/src/Volo.Abp.Data/Volo/Abp/Data/AbpCommonDbProperties.cs [hunk 3]
```diff
@@ -14,5 +14,13 @@ public static class AbpCommonDbProperties
...
         /// </summary>
         public static string DbSchema { get; set; } = null;
+
+        /// <summary>
+        /// This DbNamingConvention is shared by most of the ABP modules.
+        /// Values: Default, SnakeCase, LowerCase, UpperCase, UpperSnakeCase
+        /// 
+        /// Default value: "Default".
+        /// </summary>
+        public static string DbNamingConvention = "Default";
```
Comment (maliming) [inline framework/src/Volo.Abp.Data/Volo/Abp/Data/AbpCommonDbProperties.cs:24]: Use `enum DbNamingConvention` as type will better.

**Comment by maliming** [comment]
By the way, there are some bugs in EFCore.NamingConventions. We have to wait for its official version.

https://github.com/efcore/EFCore.NamingConventions/issues?q=is%3Aissue+label%3Abug+milestone%3A5.0.0

**Comment by danvic712** [comment]
Hi, hasn't this feature been officially released? I am using version 4.2.2 now and have not found this feature yet

**Pull request closed without merging**